### PR TITLE
Transfert of code from bitbucket

### PR DESCRIPTION
**Pull request by frheault**
Simple transfer of code, update call to scripts in order to work with scilpy master.
Simplified a bit the channel structure for centroids by adding the parameters `--use_provided_centroids` making edge-case simpler to detect.

Some channel were removed since all input are trk and that references are no longer needed.
Working just like before, for robustness it should be tested independently 

[tractometry_flow_PR.zip](https://github.com/scilus/tractometry-flow/files/4886222/tractometry_flow_PR.zip)

### CODE REVIEW

**Comment by mdesco** [comment]
can we call this one tractometry_flow please

**Comment by arnaudbore** [comment]
> can we call this one tractometry_flow please

done

File: USAGE [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+Compute tractometry on each bundles/metrics
+
+USAGE
+
+nextflow run main.nf [OPTIONAL_ARGUMENTS] (--root)
+
+DESCRIPTION
+
+--root=/path/to/[root]     Root folder containg multiple subjects
```
Comment (mdesco) [inline USAGE:12]: Please update to --input

**Review by mdesco — COMMENTED** [review]
Should the bin/jq file be put in the singularity image?
This creates issues for people on different plateformes, no?

File: main.nf [hunk 2]
```diff
@@ -0,0 +1,943 @@
...
+    input:
+    set sid, bname, file(voxel_label_map) from voxel_label_map_for_volume
+
+    output:
+    set sid, val(bname), "${sid}__${bname}__volume_per_label.json" into\
+            volume_per_label_to_aggregate
+
+    script:
+    """
+    scil_bundle_volume_per_label.py\
```
[the remark this answers is not in the document]
Comment (mdesco) [reply main.nf:451]: we have to also convert $voxel_label_map  to int16 before using it by scil_compute_bundle_volume_per_label.py
scil_image_math.py convert $voxel_label_map $voxel_label_map --data_type int16 -f 

I remember doing all these changes in the bitbucket! grrr....

**Review by mdesco — APPROVED** [review]
With my above modifications, everything runs.
Here is my main.nf
https://www.dropbox.com/s/kkpfqwz0d128vzc/main.nf?dl=0

File: USAGE [hunk 3]
```diff
@@ -0,0 +1,86 @@
+Scilpy dMRI tractometry pipeline
+================================
+
+Compute tractometry on each bundles/metrics
+
+USAGE
+
+nextflow run main.nf [OPTIONAL_ARGUMENTS] (--root)
```
Comment (arnaudbore) [inline USAGE:8]:
```suggestion
nextflow run main.nf [OPTIONAL_ARGUMENTS] (--input)
```

File: main.nf [hunk 4]
```diff
@@ -0,0 +1,943 @@
...
+    input:
+    set sid, bname, file(bundle) from outlier_rejected_bundle_for_volume
+
+    output:
+    set sid, val(bname), "${sid}__${bname}__volume.json" into\
+        volume_to_aggregate
+
+    script:
+    """
+    scil_bundle_volume.py $bundle > ${sid}__${bname}__volume_raw.json
```
Comment (arnaudbore) [inline main.nf:392]:
```suggestion
    scil_compute_bundle_volume.py $bundle > ${sid}__${bname}__volume_raw.json
```

File: main.nf [hunk 5]
```diff
@@ -0,0 +1,943 @@
...
+    set sid, bname, file(bundle), file(centroid) from\
+        in_bundle_voxel_label_map
+
+    output:
+    set sid, val(bname), "${sid}__${bname}__voxel_label_map.nii.gz" into\
+        voxel_label_map_for_volume
+
+    script:
+    """
+    scil_bundle_voxel_label_map.py  $bundle $centroid ${sid}__${bname}__voxel_label_map.nii.gz
```
Comment (arnaudbore) [inline main.nf:437]:
```suggestion
    scil_compute_bundle_voxel_label_map.py  $bundle $centroid ${sid}__${bname}__voxel_label_map.nii.gz
```

File: main.nf (hunk 2 above)
Comment (arnaudbore) [inline main.nf:451]:
```suggestion
    scil_image_math.py convert $voxel_label_map $voxel_label_map --data_type int16 -f 
    scil_compute_bundle_volume_per_label.py\
```

File: main.nf [hunk 6]
```diff
@@ -0,0 +1,943 @@
...
+    set sid, val(bname), "${sid}__${bname}__mean_std_per_point_raw.json" into\
+        mean_std_per_point_for_plot
+    set sid, val(bname), "${sid}__${bname}__mean_std_per_point.json" into\
+        mean_std_per_point_to_aggregate
+
+    script:
+    density_weighting =\
+        params.mean_std_per_point_density_weighting ? '--density_weighting' : ''
+    """
+    scil_bundle_mean_std_per_point.py --sort_keys $density_weighting $bundle\
```
Comment (arnaudbore) [inline main.nf:489]:
```suggestion
    scil_compute_bundle_mean_std_per_point.py --sort_keys $density_weighting $bundle\
```

**Review by arnaudbore — COMMENTED** [review]
Here are some suggestions (using Max comments)

**Comment by mdesco** [comment]
@frheault any chance we can prioritize tractometry_flow? 4 projects waiting on this on my end. Thanks!

**Comment by frheault** [comment]
@mdesco it is ready for review and testing, I would simply say that the random coloring is pretty bad in its choice of color, but this is outside of the scope of this PR

**Comment by arnaudbore** [comment]
Please remove jq since it's already in the scilpy docker.

**Review by mdesco — CHANGES_REQUESTED** [review]
Mini demande avant de faire le merge. On peut rajouter un process de scil_convert_json_to_xlsx.py des json finaux dans Statistics? 

Nos users le demandent toujours.

Thanks

**Comment by frheault** [comment]
@mdesco I will remove JQ, let's merge and I will immediately do a new PR.
(I have to verify the converting work correctly for all json)

That's way we can start using it.

**Comment by frheault** [comment]
@arnaudbore But what if you don't want to use the docker? I never want to user the image when I am trying things.

**Review by arnaudbore — APPROVED** [review]

**Pull request merged**